src/main/java/net/fabricmc/stitch/commands/tinyv2/CommandMergeTinyV2.java

```diff
@@ -0,0 +1,198 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.stitch.commands.tinyv2;
+
+import net.fabricmc.stitch.Command;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class CommandMergeTinyV2 extends Command {
+    public CommandMergeTinyV2() {
+        super("mergeTinyV2");
+    }
+
+    @Override
+    public String getHelpString() {
+        //TODO: mappingsBlankFillOrder?
+        return "<input-a> <input-b> <output>";
+    }
+
+    @Override
+    public boolean isArgumentCountValid(int count) {
+        return count == 3;
+    }
+
+    @Override
+    public void run(String[] args) throws IOException {
+        Path inputA = Paths.get(args[0]);
+        Path inputB = Paths.get(args[1]);
+        System.out.println("Reading " + inputA);
+        TinyFile tinyFileA = TinyV2Reader.read(inputA);
+        System.out.println("Reading " + inputB);
+        TinyFile tinyFileB = TinyV2Reader.read(inputB);
+        TinyHeader headerA = tinyFileA.getHeader();
+        TinyHeader headerB = tinyFileB.getHeader();
+        if (headerA.getNamespaces().size() != 2) {
+            throw new IllegalArgumentException(inputA + " must have exactly 2 namespaces.");
+        }
+        if (headerB.getNamespaces().size() != 2) {
+            throw new IllegalArgumentException(inputB + " must have exactly 2 namespaces.");
+        }
+
+        if (!headerA.getNamespaces().get(0).equals(headerB.getNamespaces().get(0))) {
+            throw new IllegalArgumentException(
+                    String.format("The input tiny files must have the same namespaces as the first column. " +
+                                    "(%s has %s while %s has %s)",
+                            inputA, headerA.getNamespaces().get(0), inputB, headerB.getNamespaces().get(0))
+            );
+        }
+        merge(tinyFileA, tinyFileB);
+
+        TinyV2Writer.write(tinyFileA, Paths.get(args[2]));
+    }
+
+    // Mutates inputA to be the merged result
+    private void merge(TinyFile inputA, TinyFile inputB) {
+        //TODO: how to merge properties?
+
+        System.out.println("Merging " + inputA + " with " + inputB);
+        mergeNames(inputA.getHeader().getNamespaces(), inputB.getHeader().getNamespaces());
+
+        Map<String, TinyClass> inputBClassesByFirstNamespaceName = inputB.mapClassesByFirstNamespace();
+        for (TinyClass tinyClass : inputA.getClassEntries()) {
+            String sharedName = tinyClass.getClassNames().get(0);
+            TinyClass matchingClass = inputBClassesByFirstNamespaceName.get(sharedName);
+
+            if (matchingClass == null || matchingClass.getClassNames().get(1).equals("")) {
```

Comment:
just check if two names from the two namespaces are equal instead
And doesn't the old merge command use mapping parser as well? You should just merge this one into the old one because the core logic that operate on this set of tiny model did not change